### PR TITLE
dns: add dns.rrtype keyword v2

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Feature #6666

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6666) ticket: https://redmine.openinfosecfoundation.org/issues/6666

Previous PR: https://github.com/OISF/suricata/pull/10284

Describe changes:
- added code for rrtype and implemented `DetectUintData<u16>`
- SV tests are running successfully
- Added documentation for rcode in `doc/userguide/rules/dns-keywords.rst` file.
- successfully ran the following tests:
- from rust dir: make check
- from rust dir: cargo test
- from rust dir: cargo clippy --all-features
- from rust dir: rustfmt src/dns/*
- from suricata dir: ./scripts/clang-format.sh check-branch
- from suricata dir: ./scripts/clang-format.sh branch
- from suricata dir running ./src/suricata -u showed that one unit test had failed. 
- from suricata dir running ./src/suricata -u -U dns showed 7 tests passed.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1625

### CODE REVIEW

File: rust/src/dns/detect.rs [hunk 1]
```diff
@@ -114,9 +116,44 @@ pub unsafe extern "C" fn rs_dns_detect_opcode_free(ptr: *mut c_void) {
     }
 }
 
+/// Perform the DNS rrtype match.
+///
+/// 1 will be returned on match, otherwise 0 will be returned.
+#[no_mangle]
+pub extern "C" fn rs_dns_rrtype_match(
+    tx: &mut DNSTransaction, detect: &mut DetectUintData<u16>, rrtype: u16,
```
Comment (jufajardini) [inline rust/src/dns/detect.rs:124]: Seeing this failure: https://github.com/OISF/suricata/actions/runs/7726431734/job/21062732402?pr=10289#step:13:434

I think that this last parameter here should be `flags: u8`, as we see for opcode. The info for rrtype should be present in the `detect` param, from what I understand 🤔

File: src/detect-dns-rrtype.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+static void DetectDnsRrtypeFree(DetectEngineCtx *de_ctx, void *ptr)
+{
+    SCEnter();
+    if (ptr != NULL) {
+        rs_detect_u16_free(ptr);
+    }
+    SCReturn;
+}
+
+static int DetectDnsRrtypeMatch(DetectEngineThreadCtx *det_ctx, Flow *f, uint16_t flags,
```
Comment (jufajardini) [inline src/detect-dns-rrtype.c:62]: flags here has to be `uint8_t`, cf https://github.com/OISF/suricata/actions/runs/7726431734/job/21062732402?pr=10289#step:13:434 and https://github.com/OISF/suricata/blob/master/src/detect.h#L1260

File: doc/userguide/rules/dns-keywords.rst [hunk 3]
```diff
@@ -46,6 +46,29 @@ Match on DNS requests where the **opcode** is NOT 0::
 
   dns.opcode:!0;
 
+dns.rrtype
+----------
+
+This keyword matches on the **rrtype** found in the DNS message.
+
```
Comment (jufajardini) [inline doc/userguide/rules/dns-keywords.rst:53]: Could you follow the style where you indicate that this is an integer keyword, and a reference to the integer keyword documentation that was recently merged? See example here: https://github.com/OISF/suricata/blob/master/doc/userguide/rules/dhcp-keywords.rst#dhcprebinding_time

**Review by jufajardini — CHANGES_REQUESTED** [review]
Good work!

Please see inline comments :)

For the commit message, could you move the ticket reference to the last line? This is more aligned with what we recommend and see, so it's easier for the eye to find that info if it's there :)

**Comment by hadiqaalamdar** [comment]
New PR: https://github.com/OISF/suricata/pull/10306

**Pull request closed without merging**